wordpress-mobile/WordPress-Android
# Use HTTP API calls to trigger release jobs

This PR updates our release tools to trigger any CI release job via an API call instead of relying on the automatic trigger on tags.
This way, it will be easier and cleaner to handle any failure during the release build and, moreover, it'll be possible to move more steps of the releases to CI (for example, validation and lint steps that can fail by design).

Note that in order to use the release tools, users should configure a new token for CircleCI on their local environment, following these steps:

- Go to `https://app.circleci.com/settings/user/tokens`.
- Create a Personal API Token.
- Add `CIRCLE_CI_AUTH_TOKEN=<Your Token>` to your `~/.wpandroid-env.default` (you can check out the updated `fastlane/env.example` for a template).

**To test:**
Full testing is tricky because it's tight to the release cycle. Anyway, it's possible to test the critical part of it this way:

- checkout a new `release/test` branch from this one and push it.
- add a dummy commit (the build will fail, so adding a dummy commit will avoid the failure to be reported to this PR).
- `bundle install`
- Run `bundle exec fastlane trigger_release_build branch_to_build:"release/test"`
- Verify that the release build job has been started on CircleCI. The job will fail during the upload to PlayStore because the current version is already present. This is fine.
- Delete the test branch.

`Review`
I've asked the review to @JavonDavis because he is currently the release manager of WordPress Android. All the code is already in production in WPiOS, so it should be fine, but there may always be some nuances with porting that I missed, so I think that understanding the changes will help if something happens during the next release.

PR submission checklist:
